modules/assim.sequential/R/Analysis_sda_block.R

```diff
@@ -632,4 +641,116 @@ block.2.vector <- function (block.list, X, H) {
               Pf = Pf,
               mu.a = mu.a,
               Pa = Pa))
+}
+
+##' This function provides means to split large SDA analysis (MCMC) runs into separate `qsub` jobs.
+##' Including job creation, submission, and assemble.
+##' @title qsub_analysis_submission
+##' @param block.list list: MCMC configuration lists for the block SDA analysis.
+##' @param outdir character: SDA output path.
+##' @param job.per.folder numeric: number of jobs per folder.
+##' @param cores numeric: number of cpus used for parallel computaion. Default is NULL.
+##' @export
+##' 
+qsub_analysis_submission <- function(block.list, outdir, job.per.folder = 200, cores = NULL) {
+  L <- length(block.list)
+  # calculate proper folder number based on settings.
+  folder.num <- ceiling(L/job.per.folder)
+  # create folder.
+  # if we have previous outputs, remove them.
+  if (file.exists(file.path(outdir, "qsub_analysis"))) {
+    unlink(file.path(outdir, "qsub_analysis"), recursive = T)
+  }
+  # create new folder.
+  dir.create(file.path(outdir, "qsub_analysis"))
+  # loop over sub-folders.
+  folder.paths <- job.ids <- c()
+  PEcAn.logger::logger.info(paste("Submitting", folder.num, "jobs."))
+  for (i in 1:folder.num) {
```

Comment:
some systems will penalize you if you do many submissions in parallel, or only run two and then wait for those to be done, and more jobs you submit, the lower your priority in the queue. To overcome some of this I added the `modellauncher` this will create a text file with as first line the command to execute, and will have a list of folders it needs to run the command in.